add optimizing code explanation (#24375)
mdn/content: files/en-us/webassembly/c_to_wasm/index.md

   The options we've passed are slightly different this time:

   - We've specified `-o hello2.html`, meaning that the compiler will still output the JavaScript glue code and `.html`.

>    - We've specified `-O3`, it is used to optimize the code, emcc has optimization levels like any other c compiler, they include: `-O0` (no optimization), `-O1`, `-O2`, `-Os`, `-Oz`, `-Og` and `-O3`, `-O3` are good settings for release builds.

Reviewer: ```suggestion
   - We've specified `-O3`, which is used to optimize the code. Emcc has optimization levels like any other C compiler, including: `-O0` (no optimization), `-O1`, `-O2`, `-Os`, `-Oz`, `-Og`, and `-O3`. `-O3` is a good setting for release builds.
```

Author: thanks, sorry for bad english, i use google translate
